### PR TITLE
Fixes #1310: mason add command ask for overwrite if brick is found

### LINKED ISSUE #1310: feat: ask to overwrite or not for duplicate brick name
**Description**
Hi, Thanks for wonderful project.
It would be good if `mason_cli` ask to overwrite or not for duplicate brick name.

**Steps To Reproduce**
1.  install brick from brick.hub
```
mason add -g very_good_core
```
2. install another brick from local with duplicate name
```
mason add -g very_good_core --path ./
```
3. list bricks
```
mason ls -g
C:\Users\user\AppData\Local\Mason\Cache\global
└── very_good_core 0.6.1 -> registry.brickhub.dev
``` 
It doesn't overwrite with local brick. 
However, if a local brick is installed first and then a brick from brick.hub is installed later, it overwrites the existing local brick. 
Maybe bricks from brick.hub have higher priority over local bricks? It would be good if `mason_cli` would ask to overwrite or not for duplicate brick names.

For example, `the 'very_good_core' already exists. Do you want to overwrite it? (y/n)`

### PR DESCRIPTION
**Pull request by koukibadr**
## Status

**Ready for review**

## Description

This is a part of https://github.com/felangel/mason/issues/1310, "feat: ask to overwrite or not for duplicate brick name".

Changes:

- check if global mason list or local mason list contains the brick name based on add command parameters
- prompt to overwrite or not the exisiting installed brick (default is set to true)

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/mason_cli/pubspec.yaml
Comment (felangel) [inline packages/mason_cli/pubspec.yaml:1]: Can we revert the changes to the pubspec since they aren’t related to the issue?

File: packages/mason_cli/lib/src/commands/add.dart [hunk 1]
```diff
@@ -65,6 +65,19 @@ class AddCommand extends MasonCommand with InstallBrickMixin {
     }
 
     final progress = logger.progress('Installing ${brick.name}');
+
+    final masonYaml = isGlobal ? globalMasonYaml : localMasonYaml;
+    if (masonYaml.bricks.keys.contains(brick.name)) {
+      final confirmation = logger.confirm(
```
Comment (felangel) [inline packages/mason_cli/lib/src/commands/add.dart:71]: Can you add some unit tests for the new functionality?

File: packages/mason_cli/lib/src/commands/add.dart [hunk 2]
```diff
@@ -65,6 +65,19 @@ class AddCommand extends MasonCommand with InstallBrickMixin {
...
+        'Existing ${brick.name} found, you want to overwrite it?',
```
Comment (felangel) [inline packages/mason_cli/lib/src/commands/add.dart:72]:
```suggestion
        'Existing ${brick.name} found. Do you want to overwrite it?',
```

File: packages/mason_cli/pubspec.yaml
Comment (koukibadr) [reply packages/mason_cli/pubspec.yaml:1]: Okay I'll revert the pubspec changes

**Comment by felangel** [comment]
Closing in favor of https://github.com/felangel/mason/pull/1435. Thanks for the contributions, I really appreciate you taking the time! 💙
